core/rend/dx11/dx11_overlay.cpp

```diff
@@ -140,65 +187,72 @@ void DX11Overlay::draw(u32 width, u32 height, bool vmu, bool crosshair)
 			vp.MaxDepth = 1.f;
 			deviceContext->RSSetViewports(1, &vp);
 			quad.draw(vmuTextureViews[i], samplers->getSampler(false));
+			if (config::OnlyShowVMUA1)  // Only break if single VMU mode is enabled
+				break;
 		}
 	}
 	if (crosshair)
 	{
-		for (u32 i = 0; i < config::CrosshairColor.size(); i++)
+		for (size_t i = 0; i < vmuTextures.size(); i++)
 		{
-			if (!crosshairNeeded(i))
-				continue;
-			if (!xhairTexture)
-			{
-				const u32* texData = getCrosshairTextureData();
-				D3D11_TEXTURE2D_DESC desc{};
-				desc.Width = 16;
-				desc.Height = 16;
-				desc.ArraySize = 1;
-				desc.SampleDesc.Count = 1;
-				desc.Usage = D3D11_USAGE_DEFAULT;
-				desc.BindFlags = D3D11_BIND_SHADER_RESOURCE;
-				desc.Format = DXGI_FORMAT_R8G8B8A8_UNORM;
-				desc.MipLevels = 1;
 
-				if (SUCCEEDED(device->CreateTexture2D(&desc, nullptr, &xhairTexture.get())))
+			if (!vmu_lcd_status[i])
+			{
+				if (!crosshairNeeded(i))
+					continue;
+				if (!xhairTexture)
 				{
-					D3D11_SHADER_RESOURCE_VIEW_DESC viewDesc{};
-					viewDesc.Format = desc.Format;
-					viewDesc.ViewDimension = D3D11_SRV_DIMENSION_TEXTURE2D;
-					viewDesc.Texture2D.MipLevels = desc.MipLevels;
-					device->CreateShaderResourceView(xhairTexture, &viewDesc, &xhairTextureView.get());
+					const u32* texData = getCrosshairTextureData();
+					D3D11_TEXTURE2D_DESC desc{};
+					desc.Width = 16;
+					desc.Height = 16;
+					desc.ArraySize = 1;
+					desc.SampleDesc.Count = 1;
+					desc.Usage = D3D11_USAGE_DEFAULT;
+					desc.BindFlags = D3D11_BIND_SHADER_RESOURCE;
+					desc.Format = DXGI_FORMAT_R8G8B8A8_UNORM;
+					desc.MipLevels = 1;
 
-					deviceContext->UpdateSubresource(xhairTexture, 0, nullptr, texData, 16 * 4, 16 * 4 * 16);
+					if (SUCCEEDED(device->CreateTexture2D(&desc, nullptr, &xhairTexture.get())))
+					{
+						D3D11_SHADER_RESOURCE_VIEW_DESC viewDesc{};
+						viewDesc.Format = desc.Format;
+						viewDesc.ViewDimension = D3D11_SRV_DIMENSION_TEXTURE2D;
+						viewDesc.Texture2D.MipLevels = desc.MipLevels;
+						device->CreateShaderResourceView(xhairTexture, &viewDesc, &xhairTextureView.get());
+
+						deviceContext->UpdateSubresource(xhairTexture, 0, nullptr, texData, 16 * 4, 16 * 4 * 16);
+					}
 				}
-			}
 
-			auto [x, y] = getCrosshairPosition(i);
+				auto [x, y] = getCrosshairPosition(i);
 #ifdef LIBRETRO
-			float halfWidth = lightgun_crosshair_size / 2.f / config::ScreenStretching * 100.f * config::RenderResolution / 480.f;
-			float halfHeight = lightgun_crosshair_size / 2.f * config::RenderResolution / 480.f;
-			x /= config::ScreenStretching / 100.f;
+				float halfWidth = lightgun_crosshair_size / 2.f / config::ScreenStretching * 100.f * config::RenderResolution / 480.f;
+				float halfHeight = lightgun_crosshair_size / 2.f * config::RenderResolution / 480.f;
+				x /= config::ScreenStretching / 100.f;
```

Comment:
Avoid unnecessary white space changes as it makes it difficult to see the actual changes.